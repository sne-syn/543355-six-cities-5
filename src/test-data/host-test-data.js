export const hostPro = {
  avatar: `https://robohash.org/61?set=set2&size=74x74`,
  id: 5,
  isPro: true,
  name: `Janette`,
};
export const host = {
  avatar: `https://robohash.org/61?set=set2&size=74x74`,
  id: 5,
  isPro: false,
  name: `Janette`,
};
