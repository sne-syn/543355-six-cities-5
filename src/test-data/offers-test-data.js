export const offers = [{
  bedrooms: 1,
  city: {
    location: {
      latitude: 52.370216,
      longitude: 4.895168,
      zoom: 10,
    },
    name: `Paris`,
  },
  description: `Gran suite con entrada independiente y baño interior privado en el centro de Segovia. Privacidad y comodidad a tan sólo tres minutos del acueducto.Amplia y luminosa, con 2 camas de 105 (+ 1 opcional de 90), baño completo, calefacción, mininevera, microondas, vajilla, cubertería básica, Internet y un balcón grande y soleado.I speak English, Italian and Spanish`,
  goods: [`cabel TV`, `heating`, `kitchen`, `towels`, `washing machine`, `washer`, `dishwasher`, `conditioner`, `elevator`, `coffee machine`],
  host: {
    avatar: `https://robohash.org/28?set=set2&size=74x74`,
    id: 4,
    isPro: false,
    name: `Salom`,
  },
  id: 1,
  previewImage: `https://bit.ly/34LWdhj`,
  images: [`https://bit.ly/34LWdhj`, `https://bit.ly/36Ys6Wn`, `https://bit.ly/33Qpecj`, `https://bit.ly/30YVQif`, `https://bit.ly/36TgIv7`, `https://bit.ly/3lEt2DG`],
  isFavorite: true,
  isPremium: false,
  location: {latitude: 48.865610000000004,
    longitude: 2.350499,
    zoom: 5},
  maxGuests: 8,
  price: 207,
  rating: 1.4018800815887484,
  title: `La Latina- Small Charming Studio`,
  type: `room`,
}, {
  bedrooms: 3,
  city: {
    location: {
      latitude: 52.370216,
      longitude: 4.895168,
      zoom: 10,
    },
    name: `Brussels`,
  },
  description: `In full nature and very close to public transport. Green area surrounded by all kinds of sports facilities including a golf course, paddle, artificial grass soccer, pelota court. In an urbanization of villas with landscaped plots of about 500 m2`,
  goods: [`elevator`, `dishwasher`, `coffee machine`, `washer`, `washing machine`],
  host: {
    avatar: `https://robohash.org/82?set=set2&size=74x74`,
    id: 6,
    isPro: false,
    name: `Whitney`,
  },
  id: 2,
  previewImage: `https://bit.ly/34LWdhj`,
  images: [`https://bit.ly/30YVQif`, `https://bit.ly/3lEt2DG`, `https://bit.ly/36TgIv7`, `https://bit.ly/33Qpecj`, `https://bit.ly/36Ys6Wn`, `https://bit.ly/34LWdhj`],
  isFavorite: false,
  isPremium: false,
  location: {latitude: 48.865610000000004,
    longitude: 2.350499,
    zoom: 5},
  maxGuests: 5,
  price: 194,
  rating: 1.9069547983791004,
  title: `Estudio/suite completo en el centro de Segovia`,
  type: `house`,
}];
