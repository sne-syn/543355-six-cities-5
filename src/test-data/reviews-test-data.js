export const review = {
  author: `Beaulah`,
  avatar: `https://robohash.org/54?set=set2&size=54x54`,
  date: `2004-09-06T16:12:32.554Z`,
  id: 0,
  rating: 1,
  comment: `Antoni is a great host and us a wonderful home. His suggestions for places to go were killer!10/10 would stay again.`,
};

export const reviews = [{
  author: `Ione`,
  avatar: `https://robohash.org/98?set=set2&size=54x54`,
  date: `2007-10-13T16:12:32.554Z`,
  id: 1,
  rating: 4,
  comment: `There’s a stunning beach with great snorkelling just a 6min drive, a few other beautiful beaches close by too.`,
}, {
  author: `Bruno`,
  avatar: `https://robohash.org/53?set=set2&size=54x54`,
  date: `2005-08-25T16:12:32.554Z`,
  id: 2,
  rating: 3,
  comment: `Maria’s place is at a perfect location, very easy to get around with public transportation and walking distance to awesome bars and restaurants. Her place is great, very clean and has everything you need.`,
}];
