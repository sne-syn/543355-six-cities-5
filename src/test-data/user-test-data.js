export const user = {
  userAvatar: `https://robohash.org/82?set=set2&size=74x74`,
  userEmail: `trust_no_1@gmail.com`,
  userId: 1,
  userIsPro: false,
  userName: `fox_mulder`
};
