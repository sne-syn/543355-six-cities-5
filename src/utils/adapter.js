export const adaptOffer = (data) => {
  return {
    bedrooms: data[`bedrooms`],
    city: {
      location: adaptLocation(data[`city`][`location`]),
      name: data[`city`][`name`]
    },
    description: data[`description`],
    goods: data[`goods`],
    host: adaptHost(data[`host`]),
    id: data[`id`],
    images: data[`images`],
    isFavorite: Boolean(data[`is_favorite`]),
    isPremium: Boolean(data[`is_favorite`]),
    location: adaptLocation(data[`location`]),
    maxGuests: data[`max_adults`],
    previewImage: data[`preview_image`],
    price: data[`price`],
    rating: data[`rating`],
    title: data[`title`],
    type: data[`type`],
  };
};

const adaptLocation = (location) => {
  return {
    latitude: location[`latitude`],
    longitude: location[`longitude`],
    zoom: location[`zoom`]
  };
};

const adaptHost = (host) => {
  return {
    avatar: host[`avatar_url`],
    id: host[`id`],
    isPro: Boolean(host[`is_pro`]),
    name: host[`name`]
  };
};

export const adaptReview = (review) => {
  return {
    author: review[`user`][`name`],
    avatar: review[`user`][`avatar_url`],
    date: review[`date`],
    id: review[`id`],
    isPro: Boolean(review[`is_pro`]),
    rating: review[`rating`],
    comment: review[`comment`],
    userId: review[`user`][`id`]
  };
};

export const adaptOffers = (offers) => {
  return offers.map((offer) => adaptOffer(offer));
};

export const adaptReviews = (reviews) => {
  return reviews.map((review) => adaptReview(review));
};
