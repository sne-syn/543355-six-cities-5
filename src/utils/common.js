const AccomodationTypes = {
  apartment: `Apartment`,
  room: `Private Room`,
  house: `House`,
  hotel: `Hotel`
};

// add leading zero to num < 10
const addLeadingZero = (value) => {
  return (value < 10) ? `0${value}` : value;
};

const generateID = function () {
  return `_${Math.random().toString(36).substr(2, 9)}`;
};

// generate randome integer
const getRandomIntegerNumber = (min, max) => {
  return min + Math.floor(Math.random() * (max - min));
};

// generate random non-integer
const getRandomNumber = (min, max) => {
  return min + (Math.random() * (max - min));
};

// get 1 random item from arrayOfItems
const getRandomArrayItem = (array) => {
  const randomIndex = getRandomIntegerNumber(0, array.length);
  return array[randomIndex];
};

// get several random items
const getSeveralRandomItems = (array, count) => {
  let set = new Set();
  for (let i = 0; i < count; i++) {
    set.add(getRandomArrayItem(array));
  }
  return [...set];
};

// capitalize first char only
const capitalizeChar = (str) => {
  if (typeof str !== `string`) {
    return ``;
  }
  return str.charAt(0).toUpperCase() + str.slice(1);
};

export {
  AccomodationTypes,
  addLeadingZero,
  capitalizeChar,
  generateID,
  getRandomIntegerNumber,
  getRandomArrayItem,
  getRandomNumber,
  getSeveralRandomItems,
};
