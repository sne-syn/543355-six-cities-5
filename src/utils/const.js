const NEAR_PLACES_COUNT = 3;
const ComponentType = {
  CITIES: `cities`,
  NEAR: `near-places`,
  FAVORITE: `favorites`
};

const SortType = new Map();
SortType.set(`DEFAULT`, `Popular`);
SortType.set(`PRICE_LOW_HIGH`, `Price: low to high`);
SortType.set(`PRICE_HIGH_LOW`, `Price: high to low`);
SortType.set(`TOP_RATED`, `Top rated first`);

const AccomodationTypes = {
  apartment: `Apartment`,
  room: `Private Room`,
  house: `House`,
  hotel: `Hotel`
};

const MAX_RATING = 5;
const CITIES = [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];
const MONTH_NAMES = [
  `January`,
  `February`,
  `March`,
  `April`,
  `May`,
  `June`,
  `July`,
  `August`,
  `September`,
  `October`,
  `November`,
  `December`,
];

export {NEAR_PLACES_COUNT, MAX_RATING, CITIES, MONTH_NAMES, AccomodationTypes, ComponentType, SortType};
