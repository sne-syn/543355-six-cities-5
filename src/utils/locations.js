import {getRandomIntegerNumber} from '../utils/common';

export const LocationsList = {
  Amsterdam: [
    [52.3909553943508, 4.85309666406198],
    [52.369553943508, 4.85309666406198],
    [52.3909553943508, 4.929309666406198],
    [52.3809553943508, 4.939309666406198],
    [52.4006614685059, 4.88902187347412],
    [52.3763414, 4.8999733],
    [52.333889, 4.888803]
  ], Dusseldorf: [
    [51.237402, 6.797314],
    [51.239402000000005, 6.756314000000001],
    [51.248402000000006, 6.763314],
    [51.205402, 6.7613140000000005]
  ], Cologne: [
    [50.947361, 6.9799739999999995],
    [50.949361, 6.976974],
    [50.932361, 6.960974],
    [50.945361, 6.935974],
    [50.913361, 6.9509739999999995]
  ], Brussels: [
    [50.822556999999996, 4.347697],
    [50.844556999999995, 4.346697],
    [50.869557, 4.332697],
    [50.852557, 4.3376969999999995],
    [50.842557, 4.363696999999999]
  ], Hamburg: [
    [53.563341, 10.019654000000001],
    [53.574341000000004, 10.022654000000001]
  ], Paris: [
    [48.83961, 2.342499],
    [48.865610000000004, 2.350499],
    [48.84461, 2.374499],
    [48.861610000000006, 2.340499]
  ]
};

export const CoordinatesMap = {
  AMSTERDAM: {
    latitude: 52.38333,
    longitude: 4.9,
    zoom: getRandomIntegerNumber(10, 14)
  },
  DUSSELDORF: {
    latitude: 51.225402,
    longitude: 6.776314,
    zoom: getRandomIntegerNumber(10, 14)
  },
  COLOGNE: {
    latitude: 50.938361,
    longitude: 6.959974,
    zoom: getRandomIntegerNumber(11, 14)
  },
  HAMBURG: {
    latitude: 53.550341,
    longitude: 10.000654,
    zoom: getRandomIntegerNumber(10, 14)
  },
  PARIS: {
    latitude: 48.85661,
    longitude: 2.351499,
    zoom: getRandomIntegerNumber(11, 14)
  },
  BRUSSELS: {
    latitude: 50.846557,
    longitude: 4.351697,
    zoom: getRandomIntegerNumber(10, 14)
  },
};
