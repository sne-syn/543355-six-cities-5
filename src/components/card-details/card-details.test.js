import React from 'react';
import renderer from 'react-test-renderer';
import CardDetails from './card-details';
import {ComponentType} from '../../utils/const';
import {MemoryRouter} from 'react-router-dom';
import {offer, offerPremium} from '../../test-data/offer-test-data';

jest.mock(`../favorite-button/favorite-button`, () => `FavoriteButton`);
jest.mock(`../premium-mark/premium-mark`, () => `PremiumMark`);
jest.mock(`../star-bar/star-bar`, () => `StarBar`);

const noop = () => {};


describe(`<CardDetails /> test`, () => {
  it(`renders CardDetails for cities list`, () => {
    const tree = renderer
    .create(
        <MemoryRouter>
          <CardDetails
            offer={offer}
            type={ComponentType.CITIES}
            highlightedOfferID={`0`}
            resetActiveCardIdAction={noop}
            setActiveCardIdAction={noop}>
            <React.Fragment />
          </CardDetails>
        </MemoryRouter>
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`renders CardDetails for favorites list`, () => {
    const tree = renderer
    .create(
        <MemoryRouter>
          <CardDetails
            offer={offer}
            type={ComponentType.FAVORITE}/>
        </MemoryRouter>
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`renders CardDetails for near-places list`, () => {
    const tree = renderer
    .create(
        <MemoryRouter>
          <CardDetails
            offer={offer}
            type={ComponentType.NEAR}/>
        </MemoryRouter>
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`renders CardDetails with Premium-mark`, () => {
    const tree = renderer
    .create(
        <MemoryRouter>
          <CardDetails
            offer={offerPremium}
            type={ComponentType.NEAR}/>
        </MemoryRouter>
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

