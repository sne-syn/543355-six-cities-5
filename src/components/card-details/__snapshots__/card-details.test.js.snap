// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardDetails /> test renders CardDetails for cities list 1`] = `
Array [
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="/hotels/1"
      onClick={[Function]}
    >
      <img
        alt="Place image"
        className="place-card__image"
        height={200}
        src="https://bit.ly/34LWdhj"
        width={260}
      />
    </a>
  </div>,
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          207
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <FavoriteButton
        componentName="place-card"
        defaultOnValue={false}
        offerId={1}
        type="cities"
      />
    </div>
    <StarBar
      rating={1.4018800815887484}
    />
    <h2
      className="place-card__name"
    >
      <a
        href="/hotels/1"
        onClick={[Function]}
      >
        La Latina- Small Charming Studio
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      Private Room
    </p>
  </div>,
]
`;

exports[`<CardDetails /> test renders CardDetails for favorites list 1`] = `
Array [
  <div
    className="favorites__image-wrapper place-card__image-wrapper"
  >
    <a
      href="/hotels/1"
      onClick={[Function]}
    >
      <img
        alt="Place image"
        className="place-card__image"
        height={110}
        src="https://bit.ly/34LWdhj"
        width={150}
      />
    </a>
  </div>,
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          207
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <FavoriteButton
        componentName="place-card"
        defaultOnValue={false}
        offerId={1}
        type="favorites"
      />
    </div>
    <StarBar
      rating={1.4018800815887484}
    />
    <h2
      className="place-card__name"
    >
      <a
        href="/hotels/1"
        onClick={[Function]}
      >
        La Latina- Small Charming Studio
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      Private Room
    </p>
  </div>,
]
`;

exports[`<CardDetails /> test renders CardDetails for near-places list 1`] = `
Array [
  <div
    className="near-places__image-wrapper place-card__image-wrapper"
  >
    <a
      href="/hotels/1"
      onClick={[Function]}
    >
      <img
        alt="Place image"
        className="place-card__image"
        height={200}
        src="https://bit.ly/34LWdhj"
        width={260}
      />
    </a>
  </div>,
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          207
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <FavoriteButton
        componentName="place-card"
        defaultOnValue={false}
        offerId={1}
        type="near-places"
      />
    </div>
    <StarBar
      rating={1.4018800815887484}
    />
    <h2
      className="place-card__name"
    >
      <a
        href="/hotels/1"
        onClick={[Function]}
      >
        La Latina- Small Charming Studio
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      Private Room
    </p>
  </div>,
]
`;

exports[`<CardDetails /> test renders CardDetails with Premium-mark 1`] = `
Array [
  <PremiumMark
    componentName="place-card"
  />,
  <div
    className="near-places__image-wrapper place-card__image-wrapper"
  >
    <a
      href="/hotels/1"
      onClick={[Function]}
    >
      <img
        alt="Place image"
        className="place-card__image"
        height={200}
        src="https://bit.ly/34LWdhj"
        width={260}
      />
    </a>
  </div>,
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          207
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <FavoriteButton
        componentName="place-card"
        defaultOnValue={true}
        offerId={1}
        type="near-places"
      />
    </div>
    <StarBar
      rating={1.4018800815887484}
    />
    <h2
      className="place-card__name"
    >
      <a
        href="/hotels/1"
        onClick={[Function]}
      >
        La Latina - Small Charming Studio
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      Private Room
    </p>
  </div>,
]
`;
